lib/auth.js

```diff
@@ -149,30 +161,32 @@ export default class Auth {
      * @returns {Promise.<object>} current session information
      */
     async getAccessToken(forceRefresh, remainingAttempts) {
-        if (remainingAttempts === undefined) {
-            remainingAttempts = this.options.requestAttempts;
-        }
-        if (Boolean(forceRefresh) || _isExpired(this.authObject)) {
-            try {
-                remainingAttempts--;
-                this.authObject = await _requestToken(this.authObject);
-            } catch (error) {
-                if (
-                    this.options.retryOnConnectionError &&
-                    remainingAttempts > 0 &&
-                    isConnectionError(error.code)
-                ) {
-                    if (this.options?.eventHandlers?.onConnectionError) {
-                        this.options.eventHandlers.onConnectionError(error, remainingAttempts);
+        if (!(this.grant_type && this.grant_type === GRANT_TYPE_AUTHORIZATION_CODE)) {
```

Comment:
I have a preference for "positive" checks, since they are easier to read. in this case, we could check if it is a specific grant type (without null check since we set default) and if so, return early before moving to default logic

Reply:
Thanks for the comment. Fixed it.